### ISSUE TITLE
Sending Key Value of the context

### ISSUE THREAD

**Issue by nizamsp** [smooch/smooch-android#8]
I am trying to send some key, value pairs of the user context like which screen the user invoked the feedback, what the user is viewing.

I tried using the following code. But its not working. The mail/slack app never shows this value.
I thought this will work like crashlytics's key value setting :(

```
  final Map<String, Object> customProperties = new HashMap<>();
  customProperties.put("flightSearchFrom", userEnteredFrom);
  User.getCurrentUser().addProperties(customProperties);
  ConversationActivity.show(thisActivity, Intent.FLAG_ACTIVITY_NEW_TASK);
```

**Comment by jugarrit** [comment]
@nizamsp, when you add the properties, they should show up in Slack when you run the /sk !profile command.

**Comment by nizamsp** [comment]
I tried it in slack. I am not getting any custom properties. Even in mail, any key values that we have put doesn't appear.

**Comment by jugarrit** [comment]
What's in the `userEnteredForm` object?
I have tested this functionality and can confirm that it works.

**Comment by nizamsp** [comment]
Where can I get userEnteredForm object?

**Comment by jugarrit** [comment]
Objects added to custom properties must be of type `Date`, `Number` or `String`.

**Comment by nizamsp** [comment]
This issues is reproducible. The values are not getting updated.
Steps to reproduce:

Clone the following project and open it in Android Studio.
https://github.com/smooch/smooch-android/tree/master/demo/gradle

Try changing the value of Karma. It will never reflect in the mail.
I believe that the use of customProperties is not give some context values to reproduce the user's current issues.
Last seen values also not changing in all the mails I get from smooch.

Please let me know if you need more clues in reproducing the issue.

**Comment by jugarrit** [comment]
I'll take another look when I have a chance, thanks for including repro steps.

**Comment by ykostova** [comment]
@jugarrit I am experiencing the exact same issue! 

iOS sdk is working with no issues whereas Android does not update any of the custom properties. The only way I was able to achieve any update was doing the following: 
- logout
- add the properties to the user 
- initialise Smooch again ( even though it is already initialised) 
- login with this user again 
- start a conversation 

Surely the values are to update without having to go through this full sequence of steps!

**Comment by jugarrit** [comment]
@ykostova they definitely should...

Can you try simply calling `init` while providing the `userId` in the settings?

```java
Settings settings = new Settings("<token>");
settings.setUserId("<userId>");
settings.setJWT("<jwt>");
Smooch.init(this, settings);
```

VS

```java
Settings settings = new Settings("<token>");
Smooch.init(this, settings);
Smooch.login("<userId>", "<jwt>"
```

Clearly there is an issue here, I will make sure this bug gets fixed.

**Comment by ykostova** [comment]
@jugarrit Thanks for the suggestion. Substituting any login with simple re-initialisation as above did indeed work magic. 

I will keep an eye out for the final fix though. Thanks for your help!

**Comment by jugarrit** [comment]
This is fixed in version 4.0.6. Thanks for reporting :)
